src/dodal/devices/oav/pin_image_recognition/__init__.py

```diff
@@ -153,6 +153,7 @@ async def _set_triggered_tip():
             """
             async for value in observe_value(self.array_data):
                 try:
+                    await asyncio.sleep(0)
                     location = await self._get_tip_and_edge_data(value)
                     self._set_triggered_values(location)
                 except Exception as e:
```

Comment:
Would be good if the `InvalidPinException` was raised with a message as currently the log message doesn't print any information about the exception.

Reply:
In the longs, don't we see the
`LOGGER.warning(f"Failed to detect pin-tip location, will retry with next image: {e}")`
message each time this exception is raised?